experiments/opencog/DNNs/vqa_multi_dnn/netsvocabulary.py

```diff
@@ -2,14 +2,44 @@
 import torch.nn as nn
 import torch.nn.functional as F
 
-class NetsVocab(nn.Module):
+
+class INetsVocab(nn.Module):
```

Comment:
I like the idea, but it is an abstract class rather then an interface.
And modelIndexByWord is not properly encapsulated: it is written by inherited classes, but read by this class. As this field is part of the this class I would make it private and add setter to write it from inherited classes. Otherwise it will be difficult to see whether operations on this field are consistent.

Reply:
Ok, I provided setter and getter for modelIndexByWord